### PR TITLE
[proxy/redis] impr: use redis_auth_type to switch between auth types

### PR DESCRIPTION
**Pull request by antonyc**
## Problem

On Azure we need to use username-password authentication in proxy for regional redis client.

## Summary of changes

This adds `redis_auth_type` to the config with default value of "irsa". Not specifying it will enforce the `regional_redis_client` to be configured with IRSA redis (as it's done now). 
If "plain" is specified, then the regional client is condifigured with `redis_notifications`, consuming username:password auth from URI. We plan to do that for the Azure cloud.

Configuring `regional_redis_client` is required now, there is no opt-out from configuring it.

### CODE REVIEW

File: proxy/src/bin/proxy.rs [hunk 1]
```diff
@@ -357,7 +373,7 @@ async fn main() -> anyhow::Result<()> {
 
     let cancel_map = CancelMap::default();
 
-    let redis_publisher = match &regional_redis_client {
+    let redis_publisher = match &redis_notifications_client {
```
Comment (conradludgate) [inline proxy/src/bin/proxy.rs:376]: I think I am missing context around this specific change

File: proxy/src/bin/proxy.rs [hunk 2]
```diff
@@ -319,24 +322,37 @@ async fn main() -> anyhow::Result<()> {
...
+        ("irsa", _) => match (&args.redis_host, args.redis_port) {
+            (Some(host), Some(port)) => Some(
+                ConnectionWithCredentialsProvider::new_with_credentials_provider(
+                    host.to_string(),
+                    port,
+                    elasticache_credentials_provider.clone(),
+                ),
+            ),
+            (None, None) => {
+                bail!("irsa auth requires redis-host and redis-port to be set");
```
Comment (conradludgate) [inline proxy/src/bin/proxy.rs:343]: This seems to be what is failing CI as we aren't setting any redis info during the tests

**Review by conradludgate — APPROVED** [review]

File: proxy/src/bin/proxy.rs (hunk 2 above)
Comment (antonyc) [reply proxy/src/bin/proxy.rs]: Thanks!

File: proxy/src/bin/proxy.rs (hunk 1 above)
Comment (antonyc) [reply proxy/src/bin/proxy.rs]: Thank you!

**Pull request merged**